tests/test_rtu.cpp

```diff
@@ -0,0 +1,553 @@
+
+    }
+}
+
+TEST(RTUTests, test_ModbusDataContainerUint16 ) {
+
+    using namespace everest::modbus;
+
+    ModbusDataContainerUint16 big_endian    ( ByteOrder::BigEndian,    { 0x00ff, 0x2442 } );
+    ModbusDataContainerUint16 little_endian ( ByteOrder::LittleEndian, { 0xff00, 0x4224 } );
+
+    {
+        DataVectorUint8 res = big_endian.get_payload_as_bigendian();
+        ASSERT_EQ( res.size() , big_endian.size() * 2 ); // number of elements in container
+        ASSERT_EQ( res[0] , 0xff );
+        ASSERT_EQ( res[1] , 0x00 );
+        ASSERT_EQ( res[2] , 0x42 );
+        ASSERT_EQ( res[3] , 0x24 );
+    }
+
+    {
+        DataVectorUint8 res = little_endian.get_payload_as_bigendian();
+        ASSERT_EQ( res.size() , little_endian.size() * 2 );// number of elements in container
+        ASSERT_EQ( res[0] , 0xff );
+        ASSERT_EQ( res[1] , 0x00 );
+        ASSERT_EQ( res[2] , 0x42 );
+        ASSERT_EQ( res[3] , 0x24 );
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// test serial connection helper stuff
+
+
+TEST(RTUTestHardware, test_lowlevel_serial ) {
+
+    using namespace everest::connection;
+
+    ASSERT_THROW( SerialDeviceConfiguration("/dev/does_not_exist") , std::runtime_error );
+
+    SerialDeviceConfiguration serial_device_config( "/dev/ttyUSB0" );
+
+    serial_device_config
+        .set_sensible_defaults();
+        // .set_baud_rate(SerialDeviceConfiguration::BaudRate::Baud_19200);
+        // .set_stop_bits(SerialDeviceConfiguration::StopBits::One )
+        // .set_parity( SerialDeviceConfiguration::Parity::None )
+        // .set_data_bits( SerialDeviceConfiguration::DataBits::Bit_8 );
+
+    using CFlagType = decltype( termios::c_cflag );
+
+    ASSERT_EQ( cfgetispeed( &serial_device_config.m_tty_config ), static_cast<speed_t>( B19200 ) );
+    ASSERT_EQ( cfgetospeed( &serial_device_config.m_tty_config ), static_cast<speed_t>( B19200 ) );
+
+    ASSERT_EQ( serial_device_config.m_tty_config.c_cflag & CS8    , static_cast<CFlagType> ( CS8    )); // 8 databits
+    ASSERT_EQ( serial_device_config.m_tty_config.c_cflag & CSTOPB , static_cast<CFlagType> ( 0 )); // stop bits
+    ASSERT_EQ( serial_device_config.m_tty_config.c_cflag & PARENB , static_cast<CFlagType> ( PARENB )); // parity
+
+    SerialDevice serial_device( serial_device_config );
+
+    ASSERT_EQ( serial_device.get_serial_device_config().m_tty_config, serial_device_config.m_tty_config );
+
+    // PLAN: this will be moved into a method in the connection object
+    // int serial_port = serial_device.openSerialDevice("/dev/ttyUSB0" ); // throws std::runtime_error if open fails
+    // termios tty_config;
+    // serial_device.getConfiguration( serial_port, &tty_config );
+    // serial_device.setDefaultConfiguration( &tty_config /*, options */ );
+    // serial_device.updateTimeoutConfiguration( &tty_config, 50 );
+    // serial_device.configureDevice( serial_port, &tty_config );
```

Comment:
Can all the above comments be removed or is there still something planned for the future?

Reply:
will be removed